src/transactions/feed/queryHelper.ts

```diff
@@ -228,30 +332,44 @@ async function queryTransactionsFeed(
     },
     body: JSON.stringify({
       query: TRANSACTIONS_QUERY,
-      variables: { address, localCurrencyCode, afterCursor },
+      variables: { address, localCurrencyCode, chain, afterCursor },
     }),
   })
-  return response.json()
+  const body = (await response.json()) as QueryResponse
+  // Augment the transactions with chain data, since this is not included by default
+  // from blockchain-api and is needed throughout the app.
+  return {
+    ...body,
+    data: {
+      ...body.data,
+      tokenTransactionsV3: {
+        ...body.data.tokenTransactionsV3,
+        transactions: body.data.tokenTransactionsV3.transactions.map((tx) => {
+          return { ...tx, chain }
+        }),
+      },
+    },
+  }
 }
 
 export const TRANSACTIONS_QUERY = `
-  query UserTransactions($address: Address!, $localCurrencyCode: String, $afterCursor: String) {
-    tokenTransactionsV2(address: $address, localCurrencyCode: $localCurrencyCode, afterCursor: $afterCursor) {
+  query UserTransactions($address: Address!, $localCurrencyCode: String, $afterCursor: String, $chain: Chain!) {
+    tokenTransactionsV3(address: $address, localCurrencyCode: $localCurrencyCode, afterCursor: $afterCursor, chain: $chain) {
```

Comment:
With this change, we can no longer make any backwards incompatible change to tokenTransactionsV3 right?

Reply:
Yeah, that's correct -- if we want to be safe, we can re-use the feature gate that gates showing multichain transactions to switch between the v2 and v3 endpoints.